src/jit/valuenum.cpp

```diff
@@ -6828,6 +6976,19 @@ void Compiler::fgValueNumberTree(GenTree* tree)
             // Now that we've labeled the assignment as a whole, we don't care about exceptions.
             rhsVNPair = vnStore->VNPNormalPair(rhsVNPair);
 
+            // Record the exeception set for this 'tree' in vnExcSet.
+            // First we'll record the exeception set for the rhs and
+            // later we will union in the exeception set for the lhs
+            //
+            ValueNum vnExcSet = ValueNumStore::VNForEmptyExcSet();
```

Comment:
Wouldn't be better to change `VNUnpackExc` to always initialize `pvnx` rather than doing this all other the place?

Reply:
I will think about that

Reply:
Opened #20185